### ISSUE TITLE
A11y critical violation on Select component

### ISSUE THREAD

**Issue by EmmaOKeeffe** [Workday/canvas-kit#2456]
## 🐛 Bug Report
There is a Accessibility Violation on the `Select` component - [Ensures all ARIA attributes have valid values](https://dequeuniversity.com/rules/axe/4.8/aria-valid-attr-value?application=AxeChrome)
Due to this our use of the component our A11y tests are failing due to this violation as we run the axe tests in our unit tests. Below is an image of the violation using Axe-DevTools on the Select Basic component
![Screenshot 2023-12-11 at 11 52 18](https://github.com/Workday/canvas-kit/assets/21128075/75b479d9-f7c8-42bd-ba0a-bdf5abf53939)

## To Reproduce
Have the Axe DevTools Browser extension installed.
1. Open browser extension in developer tools and have select storybook open.
2. Click 'Scan part of my page' and hover over select component in storybook until a purple dashed border is surrounding component and click.
3. Click 'Scan'.
4. Observe critical violation in results

## Expected Behavior
No violation

## Actual Results
Violation - Invalid ARIA attribute value: aria-activedescendant="rwyd1--e-mail"

## Browser (if applicable)
Chrome

## Link to repl or repo (highly encouraged)
N/A

### Error Output
[Ensures all ARIA attributes have valid values](https://dequeuniversity.com/rules/axe/4.8/aria-valid-attr-value?application=AxeChrome)
`Invalid ARIA attribute value: aria-activedescendant="rwyd1--e-mail"`

**Comment by NicholasBoll** [comment]
@EmmaOKeeffe Thank you for the issue report!

The w3c specification isn't very clear on correct handling of this use-case of `aria-activedescendant`. It is necessary to tell screen readers the intended focused element when browser focus cannot be the same as AT focus, specifically mentioned is `role=[combobox]`: https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#kbd_focus_activedescendant

Given `axe-core` says this is a violation, my guess is the `aria-activedescendant` is not pointing to a valid DOM element _while the combobox popup menu is hidden_.

What is the correct course of action given the lack of direction from the specification?

1. Remove `aria-activedescendant` while the combobox popup menu is hidden
2. Keep the popup in the DOM so `aria-activedescendant` is pointing to a valid DOM element even when the popup menu is hidden

If the choice is between to two, I highly recommend the first. Popups work by delaying initialization until a user interacts with a popup target. This is for performance. The more work is delayed until actually needed, the faster a screen shows up.

**Comment by accdc** [comment]
Hi, in the case where the popup content is hidden or removed, it is equivalent to either remove the aria-activedescendant attribute or set it to "", both will accomplish the same thing.
